modules/caddyhttp/encode/gzip/gzip_precompressed.go

```diff
@@ -0,0 +1,29 @@
+package caddygzip
+
+import (
+	"github.com/caddyserver/caddy/v2"
+	"github.com/caddyserver/caddy/v2/modules/caddyhttp/encode"
+)
+
+func init() {
+	caddy.RegisterModule(GzipPrecompressed{})
+}
+
+// GzipPrecompressed provides the file extension for files precompressed with gzip encoding
+type GzipPrecompressed struct {
```

Comment:
Maybe we should just reuse the `Gzip` type, i.e. put the methods on that type instead of making a new one.

Same for zstd.

Reply:
So just `GzipPrecompressed` as alias for `Gzip` and register it under the `http.precompressed.*` namespace?

Reply:
I suppose the same struct can be in two separate module namespaces :thinking:

Reply:
Yeah I think this can be done. I don't think you need a type named `GzipPrecompressed` at all. Just give `Gzip` the methods it needs to satisfy the interface and register it as an `http.precompressed` module. I'll be surprised if this doesn't work.

Reply:
`Gzip` still has to be register under its encoder namespace `http.encoder.gzip`. This done via `CaddyModule() caddy.ModuleInfo` (https://github.com/caddyserver/caddy/blob/master/modules/caddyhttp/encode/gzip/gzip.go#L37-L43)
How would I register the same module again with a different namespace? Wouldn't I need a different implementation of `CaddyModule()`?

Reply:
Right, I guess not then 😅

Reply:
Oops, you're right. I forgot that module info goes both ways :man_facepalming:  A module registers a type to its name, but also needs to be able to return its name given its type. So I guess you're right, a single type can't be two modules. Not a big deal, thankfully.

Reply:
TIL: just aliasing like
```go
type GzipPrecompressed Gzip
```
does not mean the aliasing type implements the same interfaces as the aliased type.
So an interface guard like this
```
var _ encode.Precompressed = (*GzipPrecompressed)(nil)
```
would does not compile

Reply:
> TIL: just aliasing like
> 
> ```go
> type GzipPrecompressed Gzip
> ```
> 
> does not mean the aliasing type implements the same interfaces as the aliased type.

This is an alias, rather it's a new type definition. The example in [the "Alias declarations" and "Type definition" sections of the spec](https://golang.org/ref/spec#Type_declarations) show the distinction between the two.